src/main/kotlin/org/pkl/lsp/ast/Expr.kt

```diff
@@ -380,8 +370,10 @@ class PklSubscriptExprImpl(
   override val parent: PklNode,
   override val ctx: Node,
 ) : AbstractPklNode(project, parent, ctx), PklSubscriptExpr {
-  override val leftExpr: PklExpr by lazy { ctx.children[0].toNode(project, this) as PklExpr }
-  override val rightExpr: PklExpr by lazy { ctx.children[2].toNode(project, this) as PklExpr }
```

Comment:
This was the source of some bugs: we already initialize a node's children, but then initialize them yet again with these `toNode()` calls. This means that we have duplicate representations, which is both a memory/cpu waste, and also was messing with our resolvers, which use `===` to compare nodes.